src/systems/track_controller/TrackController.cc

```diff
@@ -268,6 +281,26 @@ void TrackController::Configure(const Entity &_entity,
   gzdbg << "Subscribed to " << corTopic << " for receiving track center "
          << "of rotation commands." << std::endl;
 
+  // Publish track odometry
+  const auto kDefaultOdometryTopic = topicPrefix + "/odometry";
+  const auto odometryTopic = validTopic({_sdf->Get<std::string>(
+    "odometry_topic", kDefaultOdometryTopic).first, kDefaultOdometryTopic});
+  this->dataPtr->odometryPub =
+    this->dataPtr->node.Advertise<msgs::Odometry>(odometryTopic);
+
+  double odometryFreq = _sdf->Get<double>(
+    "odometry_publish_frequency", 50).first;
+  std::chrono::duration<double> odometryPer{0.0};
+  if (odometryFreq > 0)
+  {
+    odometryPer = std::chrono::duration<double>(1 / odometryFreq);
+    this->dataPtr->odometryPubPeriod =
+      std::chrono::duration_cast<std::chrono::steady_clock::duration>(odometryPer);
+  }
+  gzdbg << "Publishing odometry to " << odometryTopic
```

Comment:
Can we document this topic in the header?

Reply:
https://github.com/gazebosim/gz-sim/pull/2021/commits/ef405008ad8d92e98bca3cd1ea43127b8bdda3ee